utils/biospecimen.js

```diff
@@ -625,8 +625,13 @@ const biospecimenAPIs = async (req, res) => {
         if(Object.keys(requestData).length === 0 ) return res.status(400).json(getResponseJSON('Request body is empty!', 400));
         const { addKitAssemblyData } = require('./firestore');
         const response = await addKitAssemblyData(requestData);
-        if(!response) return res.status(404).json(getResponseJSON('ERROR!', 404));
-        return res.status(200).json({message: `Success!`, code:200})
+        if(!response) {
+            return res.status(404).json({data: response, code:404});
+        } else if (response instanceof Error) {
+            return res.status(500).json({data: response, code: 500});
+        } else {
+            return res.status(200).json({message: `Success!`, code:200})
+        }
```

Comment:
Thanks for all the consideration about error handling. This type of work will definitely strengthen our codebase and it's really appreciated.

This can still hit an error prior to the ones you're catching in addKitAssemblyData() (like a db connection error). So please consider this pattern (very clean once you remove my notes). You can use this pattern for dependable error handling throughout all the POST requests in this PR (assuming the functions in the calling app are ready for the format):
```javascript
// Endpoint
try {
    const { addKitAssemblyData } = require('./firestore');
    const response = await addKitAssemblyData(requestData);
    return res.status(200).json({message: `Success!`, code:200})
    // There is no false path here, we're either returning true or throwing an error (Please see updated function)
    // if(!response) {
    //    return res.status(404).json({data: response, code:404});
    //}
    // Please see the updated function. We don't have to handle the instanceof Error anymore because we're throwing it to the catch block
    // } else if (response instanceof Error) {
    //    return res.status(500).json({data: response, code: 500});
    // } else { // Else is no longer needed, it will default to the successful return
    // }
} catch (error) {
    console.error(error);
    return res.status(500).json(getResponseJSON(error.message, 500)); // This message was set during the throw operation
        // or return res.status(500).json(getResponseJSON('Internal Server Error', 500));
}

// Function:
const addKitAssemblyData = async (data) => {
    try {
        await db.collection('kitAssembly').add(data);
        return true;
    }
    catch(error){
        console.error(error); // log the actual error
        throw new Error('<YOUR_CUSTOM_ERROR_MESSAGE>', { cause: error }); //throw the error back to the endpoint (I think Warren prefers adding the cause because it holds more error details if we want to access them in the future).
        // You can also just 'throw error', but this risks a less understandable error for the user and risks exposing sensitive data depending on the use.
    }
}
```